cmd/geth/les_test.go

```diff
@@ -166,6 +166,7 @@ func TestPriorityClient(t *testing.T) {
 		freeCli.getNodeInfo().ID:     freeCli,
 		prioCli.getNodeInfo().ID:     prioCli,
 	}
+	time.Sleep(1 * time.Second)
```

Comment:
Perhaps to move this sleeping after here https://github.com/ethereum/go-ethereum/pull/21410/files#diff-2fa5be81f9d5e118c5960933593a41c0R157

Then we can ensure the previous client is kicked out, the new client is accepted.

Reply:
Myeah, maybe if this doesn't fix it. In general, it seems pretty flakey. For example, I fixed the `startGeth` to not blindly wait, but instead poll until the ipc socket opened up -- the test ran much faster, but also failed a lot more. 
It would be nice to have some better guarantees about syncronicity, but for now we have these sleeps on odd places.